src/Service/Resolvers/Sql Query Structures/BaseSqlQueryStructure.cs

```diff
@@ -506,5 +482,62 @@ public void ProcessOdataClause(FilterClause odataClause)
         {
             return filterClause.Expression.Accept<string>(visitor);
         }
+
+        /// <summary>
+        /// Gets the value of the parameter cast as the system type
+        /// </summary>
+        /// <param name="fieldValue">Field value as a string</param>
+        /// <param name="fieldName">Field name whose value is being converted to the specified system type. This is used only for constructing the error messages incase of conversion failures</param>
+        /// <param name="systemType">System type to which the parameter value is parsed to</param>
+        /// <returns>The parameter value parsed to the specified system type</returns>
+        /// <exception cref="DataApiBuilderException">Throws a DataApiBuilderException when the conversion of parameter value to the specified system type fails. The error message returned will be different in development
+        /// and production modes. In production mode, the error message returned will be generic so as to not reveal information about the database object backing the entity</exception>
+        protected object GetParamAsSystemType(string fieldValue, string fieldName, Type systemType)
+        {
+            try
+            {
+                return ParseParamAsSystemType(fieldValue, systemType);
+            }
+            catch (Exception e) when (e is FormatException || e is ArgumentNullException || e is OverflowException)
+            {
+
+                string errorMessage;
+                if (MetadataProvider.IsDevelopmentMode())
+                {
+                    if (MetadataProvider.EntityToDatabaseObject[EntityName].SourceType is SourceType.StoredProcedure)
+                    {
+                        errorMessage = $@"Parameter ""{fieldValue}"" cannot be resolved as stored procedure parameter ""{fieldName}"" " +
+                                $@"with type ""{systemType.Name}"".";
+                    }
+                    else
+                    {
+                        errorMessage = $"Parameter \"{fieldValue}\" cannot be resolved as column \"{fieldName}\" " +
+                                $"with type \"{systemType.Name}\".";
+                    }
+                }
+                else
+                {
+                    string fieldNameToBeDisplayedInErrorMessage = fieldName;
+
+                    if (MetadataProvider.EntityToDatabaseObject[EntityName].SourceType is SourceType.Table || MetadataProvider.EntityToDatabaseObject[EntityName].SourceType is SourceType.View)
```

Comment:
Instead of accessing the dictionary `EntityToDatabaseObject` twice here and once above, can use a variable to store the source type and use that variable then.

Reply:
Thanks for the suggestion. Updated